ui/src/tool-panel/tool-panel.tsx

```diff
@@ -328,53 +333,55 @@ const ToolPanel: React.FC<ToolPanelProps> = ({
             } as ToolPanelTab;
         });
 
-        const uniqueGeometrySwitches = [
-            ...suggestedSwitches.map((s) => ({
-                id: s.id,
-                name: s.name,
-                switchId: s.geometrySwitchId,
-                planId: s.geometryPlanId,
-                suggestedSwitch: s,
-            })),
-            // TODO: GVT-826 We could avoid this awkwardness if the selection item for geometry stuff would contain the geometry ID rather than (temp) layout ID
-            ...geometrySwitchIds
-                .map((sId) => {
-                    const geomSwitch = getPlan(sId.planId)?.switches?.find((s) => s.id === sId.id);
-                    return geomSwitch
-                        ? {
-                              id: geomSwitch.sourceId as GeometrySwitchId,
-                              name: geomSwitch.name,
-                              switchId: geomSwitch.sourceId as GeometrySwitchId,
-                              planId: sId.planId,
-                              suggestedSwitch: undefined,
-                          }
-                        : undefined;
-                })
-                .filter(filterNotEmpty),
-        ].filter(filterUniqueById((s) => s.switchId));
-
-        const geometrySwitchTabs = uniqueGeometrySwitches.map((s) => {
+        const suggestedSwitchTabs: ToolPanelTab[] = suggestedSwitches.map((ss) => {
             return {
-                asset: { type: 'GEOMETRY_SWITCH', id: s.id },
-                title: s.name || '-',
+                asset: { type: 'GEOMETRY_SWITCH', id: ss.id },
+                title: ss.name || '-',
                 element: (
                     <GeometrySwitchInfobox
```

Comment:
Näille infoboxeille voisi kyllä tehdä omat containerit niin ei tarvitsisi käyttää näin monilla argumenteilla kun ne tulee kuitenkin storesta.